src/main/groovy/org/labkey/gradle/task/JspCompile2Java.groovy

```diff
@@ -18,29 +18,38 @@ package org.labkey.gradle.task
 import org.apache.commons.io.FileUtils
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
-import org.gradle.api.tasks.CacheableTask
-import org.gradle.api.tasks.InputDirectory
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.PathSensitive
-import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.TaskAction
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.tasks.*
 import org.labkey.gradle.util.BuildUtils
 
+import javax.inject.Inject
+
 @CacheableTask
-class JspCompile2Java extends DefaultTask
+abstract class JspCompile2Java extends DefaultTask
 {
     public static final String CLASSES_DIR = "jspTempDir/classes"
 
+    private FileSystemOperations fileSystemOperations
+
     @PathSensitive(PathSensitivity.RELATIVE)
     @InputDirectory
     File webappDirectory
 
     @OutputDirectory
-    File getClassesDirectory()
-    {
-        return BuildUtils.getBuildDirFile(project, CLASSES_DIR)
+    final abstract DirectoryProperty classesDirectory = project.objects.directoryProperty().convention(project.layout.buildDirectory.dir(CLASSES_DIR).get())
+
+    @InputFiles
+    @CompileClasspath
+    abstract ConfigurableFileCollection getCompileClasspath()
+
+    @Inject
+    JspCompile2Java(FileSystemOperations fs) {
+        fileSystemOperations = fs
```

Comment:
Is this functionally different from the way the file system is injected in `SchemaCompile` or just a different way to accomplish the same thing?
```
@Inject abstract FileSystemOperations getFs()
```

Reply:
It is, as far as I know, just a different way to accomplish the same thing. My first attempts to do the injection via the getter didn't seem to work, so I went with the constructor route.  I might be able to switch this, but decided to leave this in to serve as an example for my future self.